### ISSUE TITLE
Help displaying an image from file

### ISSUE THREAD

**Issue by Hacking-Away** [s00500/ESPUI#318]
This lib is great, I am only one day in and I have webpages with controls and all sorts showing up where before I was following a lot of tutorials dealing with raw HTML assembling and getting frustrated, fast......, but I feel like I must be missing something with ESPUI and images.

It seems the way to display an image is in a tag, per the docs. 

> In fact, because HTML can be used in the label's value, you can make a label display images by including an <img> tag.

That's OK for images already accessible by http, but the images I want to display are in LittleFS.

I found https://github.com/s00500/ESPUI/issues/143 which seems to build up a string from a buffer - I don't completely understand if I need to do something liek that, streaming the file from LittelFS with file.open and so on, or if - **and this is what I hope is true** - I missed something and it's built in, and I can just specify an existing file from LittleFS.

Help appreciated :-)

**Comment by Hacking-Away** [comment]
In fact, this might be a file problem more than an image one because this also doesn't work for me:

`    ESPUI.fileDisplay("Filetest", ControlColor::Turquoise, "/test.txt");
`
test.txt reads OK elsewhere in code using this:

```
File file = LittleFS.open("/test.txt", "r"); 
 Serial.println("LittleFS testfile content:");
   while(file.available()){`
  Serial.write(file.read());
  }
```

**Comment by Hacking-Away** [comment]
Well.... i am displaying images sucessfully but it seems like a hack - i set up another asyncwebserver on port 81 and set the image URL in the label to http://myhostname.local:81/imagename

My second webserver just reads images from littleFS and makes them available on port 81.

**Comment by Hacking-Away** [comment]
Could you explain how that works please?  My understanding is that the newly defined "server" will point to the same instance of the aynscwebserver object that is made by ESPUI....and then we are saying for requests to "/" to juse server the called image file.   This doesn't screw up ESPUI internally?

Thanks I will try this way in the meantime.

**Comment by zajac-grzegorz** [comment]
> Could you explain how that works please? My understanding is that the newly defined "server" will point to the same instance of the aynscwebserver object that is made by ESPUI....and then we are saying for requests to "/" to juse server the called image file. This doesn't screw up ESPUI internally?
> 
> Thanks I will try this way in the meantime.

Yes, exactly. The 'server' pointer will point to the instance of the AsyncWebServer created inside the ESPUI. Then you can attach other handlers to it, like i.e. 'servestatic'. Nothing will break.

**Comment by MartinMueller2003** [comment]
Just a note: By convention, the "/" endpoint should return an index.html file or nothing at all. You can add any other endpoints you like below that.

**Comment by raysii3** [comment]
Hi, I have tried without success by replicating the code example from @zajac-grzegorz 

Can I request for a working example(much like the [text stream example](https://github.com/MartinMueller2003/ESPUI/tree/master/pio_examples))? Either by serving local images via another port or by utilizing the EspuiLittleFS?

Thank you!
